src/views/screens/EditProfileScreen.js

```diff
@@ -0,0 +1,203 @@
+import COLORS from '../../const/colors';
+import Input from '../components/Input';
+import Loader from '../components/Loader';
+import UserContext from '../context/UserContext';
+import { LinearGradient } from 'expo-linear-gradient';
+
+const EditProfileScreen = ({ navigation }) => {
+  const { data, submitData } = useContext(UserContext);
+  const [ inputs, setInputs ] = React.useState({
+    email: data.email,
+    fullname: data.fullname,
+    phone: data.phone,
+    password: data.password,
+    instagram: '',
+    date: ''
+  });
+  const [ errors, setErrors ] = React.useState({});
+  const [loading, setLoading] = React.useState(false);
+
+  const validate = () => {
+    Keyboard.dismiss();
+    let valid = true;
+    if(!inputs.email) {
+      handleError('Please input an email', 'email');
+      valid = false;
+    } else if (!inputs.email.match(/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w\w+)+$/)) {
+      handleError('Email type not valid', 'email');
+      valid = false;
+    }
+    if(!inputs.fullname) {
+      handleError('Please input fullname', 'fullname');
+      valid = false;
+    }
+    if(!inputs.phone) {
+      handleError('Please input phone', 'phone');
+      valid = false;
+    }
+    if(!inputs.password) {
+      handleError('Please input password', 'password');
+      valid = false;
+    } else if (inputs.password.length < 8) {
+      handleError('Minimum password length of 8', 'password');
+      valid = false;
+    }
+    if (valid) {
+      register();
+    }
+  };
+
+  const register = () => {
+    setLoading(true);
+    setTimeout(() => {
+      setLoading(false);
+      try {
+        AsyncStorage.setItem("user", JSON.stringify(inputs));
+        submitData(inputs);
+        navigation.navigate('HomeScreen');
+      } catch (err) {
+        Alert.alert("Error", 'Something went wrong')
+      }
+    }, 2000);
+  };
+
+  const handleOnChange = (text, input) => {
+    setInputs((prevState) => ({ ...prevState, [input]: text }));
+  };
+  const handleError = (errorMessage, input) => {
+    setErrors((prevState) => ({ ...prevState, [input]: errorMessage }))
+  };
+
+  return (
+    <SafeAreaView style={{ backgroundColor: COLORS.white, flex: 1 }}>
+      <Loader visible={loading} />
+      <ScrollView contentContainerStyle={{ paddingTop: 70, paddingHorizontal: 20 }}>
+      <View style={{ marginVertical: 20 }}>
+        <Input 
+          value={inputs.fullname}
+          label="Fullname" 
+          type="ionicon"
+          iconName="account"
+          error={errors.fullname}
+          onFocus={() => {
+            handleError(null, 'fullname');
+          }}
+          onChangeText={text => handleOnChange(text, "fullname")}
+        />
+        <Input 
+          keyboardType="numeric"
+          placeholder="Enter Birth date" 
+          label="Birth date" 
+          iconName="calendar"
+          error={errors.date}
+          onFocus={() => {
+            handleError(null, 'date');
```

Comment:
you can put it in one line: `onFocus={() => handleError(null, 'date')}`

Reply:
same for other function, when you don't have multiple functions, you can simply return without {}